src/databricks/labs/ucx/source_code/files.py

```diff
@@ -44,11 +50,93 @@ def __repr__(self):
         return f"<LocalFile {self._path}>"
 
 
+class LocalDirectory(SourceContainer):
+    def __init__(self, path: Path, file_loader: FileLoader, dir_loader: DirectoryLoader):
+        self._path = path
+        self._file_loader = file_loader
+        self._dir_loader = dir_loader
+
+    @property
+    def path(self) -> Path:
+        return self._path
+
+    def build_dependency_graph(self, parent: DependencyGraph) -> list[DependencyProblem]:
+        return list(self._build_dependency_graph(parent))
+
+    def _build_dependency_graph(self, parent: DependencyGraph) -> Iterable[DependencyProblem]:
+        for child_path in self._path.iterdir():
+            if child_path.is_dir():
+                dependency = Dependency(self._dir_loader, child_path)
+                yield from parent.register_dependency(dependency).problems
+                continue
+            dependency = Dependency(self._file_loader, child_path)
+            yield from parent.register_dependency(dependency).problems
+
+    def __repr__(self):
+        return f"<LocalDirectory {self._path}>"
+
+
+class LocalFilesLinter:
+
+    def __init__(
+        self,
+        languages: Languages,
+        file_loader: FileLoader,
+        dir_loader: DirectoryLoader,
+        path_lookup: PathLookup,
+        dependency_resolver: DependencyResolver,
+    ) -> None:
+        self._languages = languages
+        self._file_loader = file_loader
+        self._dir_loader = dir_loader
+        self._path_lookup = path_lookup
+        self._dependency_resolver = dependency_resolver
+        self._extensions = {".py": Language.PYTHON, ".sql": Language.SQL}
+
+    def lint(self, path: Path) -> list[LocatedAdvice]:
+        return list(self._lint(path))
```

Comment:
what does `databricks labs ucx lint-local-code` show?..

Reply:
output results in the following format, so that it's navigatable with CMD+Click into major IDEs:

```
relative/path/to/file1.py:START_LINE [message-code] full message
relative/file2.py:START_LINE [another-message-code] another full message
```

see:
<img width="795" alt="image" src="https://github.com/databrickslabs/ucx/assets/259697/edd0166a-b526-4dc2-8c0d-2a28af72ac5e">
<img width="714" alt="image" src="https://github.com/databrickslabs/ucx/assets/259697/1272cc50-22fc-4b89-970e-673e55e7e936">

Reply:
done

Reply:
not done.